File: src/main/java/com/musinsa/subject/mapper/ProductMapper.java
package com.musinsa.subject.mapper;

import com.musinsa.subject.entity.Product;
import com.musinsa.subject.model.product.ProductResponse;
import org.mapstruct.InjectionStrategy;
import org.mapstruct.Mapper;
import org.mapstruct.ReportingPolicy;

@Mapper(
        componentModel = "spring",
        injectionStrategy = InjectionStrategy.CONSTRUCTOR,
        unmappedTargetPolicy = ReportingPolicy.IGNORE
)
public interface ProductMapper {

    ProductResponse toResponseProduct(Product source);

}

File: src/main/kotlin/com/musinsa/subject/mapper/ProductMapper.kt
package com.musinsa.subject.mapper

import com.musinsa.subject.entity.Product
import com.musinsa.subject.model.product.ProductResponse
import io.swagger.v3.oas.annotations.Hidden
import org.mapstruct.InjectionStrategy
import org.mapstruct.Mapper
import org.mapstruct.ReportingPolicy

@Hidden
@Mapper(
    componentModel = "spring",
    injectionStrategy = InjectionStrategy.CONSTRUCTOR,
    unmappedTargetPolicy = ReportingPolicy.IGNORE
)
interface ProductMapper {
    fun toResponseProduct(source: Product): ProductResponse
}
